org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -0,0 +1,236 @@
+import static org.eclipse.scanning.sequencer.nexus.SolsticeConstants.GROUP_NAME_SOLSTICE_SCAN;
+import static org.eclipse.scanning.sequencer.nexus.SolsticeConstants.PROPERTY_NAME_UNIQUE_KEYS_PATH;
+
+import java.io.File;
+import java.util.List;
+
+import org.eclipse.dawnsci.nexus.INexusDevice;
+import org.eclipse.dawnsci.nexus.NXcollection;
+import org.eclipse.dawnsci.nexus.NexusNodeFactory;
+import org.eclipse.dawnsci.nexus.NexusScanInfo;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectProvider;
+import org.eclipse.dawnsci.nexus.builder.NexusObjectWrapper;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.ILazyWriteableDataset;
+import org.eclipse.january.dataset.IntegerDataset;
+import org.eclipse.january.dataset.LazyWriteableDataset;
+import org.eclipse.january.dataset.SliceND;
+import org.eclipse.scanning.api.AbstractScannable;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.models.ScanModel;
+import org.eclipse.scanning.api.scan.rank.IScanRankService;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
+import org.eclipse.scanning.sequencer.ServiceHolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The scan points writer creates and writes to the unique keys and points
+ * datasets in a nexus file. The unique keys dataset can be used to track how far the scan has
+ * progressed.
+ * @author Matthew Dickie
+ */
+public class SolsticeScanMonitor extends AbstractScannable<Object> implements INexusDevice<NXcollection> {
+	
+	private static final Logger logger = LoggerFactory.getLogger(SolsticeScanMonitor.class);
+
+	// Nexus
+	private List<NexusObjectProvider<?>> nexusObjectProviders = null;
+	private NexusObjectWrapper<NXcollection> nexusProvider = null;
+	
+	// Writing Datasets
+	private ILazyWriteableDataset uniqueKeys = null;
+	private ILazyWriteableDataset scanFinished = null;
+
+	// State
+	private boolean malcolmScan = false;
+	private final ScanModel model;
+	
+	public SolsticeScanMonitor(ScanModel model) {
+		this.model = model;
+	}
+
+	public void setNexusObjectProviders(List<NexusObjectProvider<?>> nexusObjectProviders) {
+		this.nexusObjectProviders = nexusObjectProviders;
+	}
+	
+	public void setMalcolmScan(boolean malcolmScan) {
+		this.malcolmScan = malcolmScan;
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.eclipse.dawnsci.nexus.INexusDevice#getNexusProvider(org.eclipse.dawnsci.nexus.NexusScanInfo)
+	 */
+	@Override
+	public NexusObjectProvider<NXcollection> getNexusProvider(NexusScanInfo info) {
+		
+		// Note: NexusScanFileManager relies on this method returning the same object each time
+		if (nexusProvider == null) {
+			NXcollection scanPointsCollection = createNexusObject(info);
+			nexusProvider = new NexusObjectWrapper<NXcollection>(GROUP_NAME_SOLSTICE_SCAN, scanPointsCollection);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)